packages/compose/src/higher-order/if-condition/index.tsx

```diff
@@ -1,8 +1,13 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { ComponentType } from 'react';
```

Comment:
Instead of telling ESLint to shut up, how about we export that type from our package instead?

Reply:
Not really telling it to "shut up" 😆 

There are almost ninety other instances of disabling this error for a type import. I'm happy to try and get back to adjust `@wordpress/element` but it's still a JavaScript file without exported types. I'd prefer to not hold up this PR while waiting to change `@wordpress/element` and update all of those other existing cases.

Thoughts?

Reply:
It's disabling this rule:
https://github.com/WordPress/gutenberg/blob/cc569b28ab7612b0bd530cc53872677374553750/.eslintrc.js#L84-L88

It was discussed several times before. Ideally, we would re-import APIs we use from `@wordpress/element` for consistency with how React is used. However, it never seemed to be a simple task with TypeScript. People started disabling this rule when importing rules. We should instead seek ways to update this rule so it doesn't error for types.

Reply:
It looks like it's a simple as using a different rule for TS files:

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-restricted-imports.md

Reply:
PR is ready: https://github.com/WordPress/gutenberg/pull/37862.

Reply:
rebased against new `trunk`, which contains #37862 
thanks again @gziolo !